keyboards/xo87/rgb/rgb.c

```diff
@@ -0,0 +1,95 @@
+/* Copyright 2021 datafx
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "rgb.h"
+#include "rgb_matrix_types.h"
+#include <print.h>
+#include "raw_hid.h"
+#include "usb_descriptor.h"
+#include "rgb_matrix.h"
```

Comment:
Pretty sure most of these includes aren't required at all. I wouldn't be surprised if they can all be replaced by:
```suggestion
#include "quantum.h"
```

Otherwise, remove the unnecessary ones -- you aren't touching the usb descriptors, raw hid is disabled in your rules.mk, and rgbmatrix likely includes its own types already.

Reply:
You are correct, thanks!